### ISSUE TITLE
C++ API neural net examples

### ISSUE THREAD

**Issue by beniz** [tensorflow/tensorflow#85]
From looking at the code, it seems most NN operations are of course available at C++ level (I couldn't find dropout, but I guess it is formalized as one of the randomized operations).

The C++ documentation outside the code itself is rather scarce however, which I fully understand for a first release, with priority given to Python.

I wouldn't be surprised that some Googlers are actually working from the straight C++ API. Is there any plan to provide some more straightforward toy applications in C++ ? If not, please just ignore and close this issue. 

Otherwise, I am interested in any pointers and I can help rewriting stuff such as seq2seq and rnn from the C++ point of view. One of the useful cornerpoints to look at could be the intersection of the Python API with the underlying C++ engine, but I haven't been able to locate it clearly yet.

**Comment by jendap** [comment]
The usual usage of tensorflow is that people develop the algorithms in python. It is much easier. Plus you can visualize the results and what not.

The deployment is typically done by taking calling a few functions and using algorithm serialized in proto. See the android example. Minimal interface ([TensorflowClassifier.java](/tensorflow/tensorflow/blob/master/tensorflow/examples/android/src/org/tensorflow/demo/TensorflowClassifier.java)) and one protobuf blob ([tensorflow_inception_graph.pb](/tensorflow/tensorflow/blob/master/tensorflow/examples/android/assets/tensorflow_inception_graph.pb)).

People are really doing it this way. It is easier. But of course it is completely fine doing all in C++. Look for the macros "REGISTER_*". I found those useful to see understand the interaction with python.

**Comment by beniz** [comment]
@jendap very useful thank you, especially the jni binder file [tensorflow_jni.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/android/jni/tensorflow_jni.cc). Hopefully it may help others who may be looking to bypass Python and directly connect to the engine.

**Comment by beniz** [comment]
I definitely got the flow better, here is the relevant doc http://tensorflow.org/api_docs/cc/index.md that I missed on the first lecture.

**Comment by raginisharma14** [comment]
@beniz , Not able to open the link you provided. http://tensorflow.org/api_docs/cc/index.md
